py/formak/python.py

```diff
@@ -66,8 +67,18 @@ def model(self, dt, state, control_vector=None):
         assert control_vector.shape == (self.control_size, min(self.control_size, 1))
 
         next_state = np.zeros((self.state_size, 1))
-        for i, impl in enumerate(self._impl):
-            next_state[i, 0] = impl(dt, *state, *control_vector)
+        for i, (state_id, impl) in enumerate(zip(self.arglist_state, self._impl)):
```

Comment:
Try refactoring for out of band error handling here in the fail pass-through case